shuup/utils/multilanguage_model_form.py

```diff
@@ -75,7 +75,7 @@ def __init__(self, **kwargs):  # noqa (C901)
             for lang in self.languages:
                 language_field = copy.deepcopy(base)
                 language_field_name = "%s__%s" % (f.name, lang)
-                if language_field.required:
+                if language_field.required and lang in self.required_languages:
```

Comment:
check the line 80, it has the same logic you are just adding. shouldn't you just move it to the line before the `if` ?

Reply:
actually, where is this `required_translated_field_names` even used?